sourced/ml/algorithms/uast_struct_to_bag.py

```diff
@@ -87,103 +100,77 @@ def __init__(self, p_explore_neighborhood, q_leave_neighborhood, n_walks, n_step
         self.q_leave_neighborhood = q_leave_neighborhood
         self.n_walks = n_walks
         self.n_steps = n_steps
-        self.type2ind = type2ind if type2ind is not None else FakeVocabulary()
+        self.node2index = node2index if node2index is not None else Node2InternalType()
+        if seed is not None:
+            random.seed(seed)
 
-    def uast2walks(self, uast):
+    def __call__(self, uast):
         starting_nodes = self.prepare_starting_nodes(uast)
-        res = []
         for i in range(self.n_walks):
             for start_node in starting_nodes:
-                res.append(self.random_walk(start_node))
-        return res
+                yield self.random_walk(start_node)
 
     @staticmethod
     def _extract_node(node, parent):
         return Node(parent=parent, internal_type=node.internal_type)
 
     def prepare_starting_nodes(self, uast):
         starting_nodes = []
-        self._prepare_starting_nodes(uast, None, starting_nodes)
-        return starting_nodes
-
-    def _prepare_starting_nodes(self, root, parent, starting_nodes):
-        node = self._extract_node(node=root, parent=parent)
-        starting_nodes.append(node)
-
-        for ch in root.children:
-            node.children.append(self._prepare_starting_nodes(
-                ch, parent=node, starting_nodes=starting_nodes))
+        root = self._extract_node(uast, None)
+        stack = [(root, uast)]
+        while stack:
+            parent, parent_uast = stack.pop()
+            children_nodes = [self._extract_node(child, parent) for child in parent_uast.children]
+            parent.children = children_nodes
+            stack.extend(zip(children_nodes, parent_uast.children))
+            starting_nodes.append(parent)
 
-        return node
+        return starting_nodes
 
     def random_walk(self, node):
         walk = [node]
         while len(walk) < self.n_steps:
             walk.append(self.alias_sample(walk))
 
-        walk = [self.node2feat(n) for n in walk]
+        walk = [self.node2index[n] for n in walk]
         return walk
 
-    def node2feat(self, node):
-        return self.type2ind[node.internal_type]
-
     def alias_sample(self, walk):
         """
         Compare to node2vec this sampling is a bit simpler because there is no loop in tree ->
         so there are only 2 options with unnormalized probabilities 1/p & 1/q
         :param walk: list of visited nodes
         :return: next node to visit
         """
-        # notation from article - t - node from previous step, v - last node
-        v = walk[-1]
+        last_node = walk[-1]
```

Comment:
https://github.com/src-d/guide/blob/master/engineering/python-style-guide.md
> Do not use single letter argument names; use X and Y only in sklearn context.

Reply:
if not one character - please make comment with mapping to variable names in the article.

Reply:
ok!